qa/rpc-tests/feature_fedpeg.py

```diff
@@ -0,0 +1,338 @@
+        assert 'vout' in raw_pegout and len(raw_pegout['vout']) > 0
+        pegout_tested = False
+        for output in raw_pegout['vout']:
+            scriptPubKey = output['scriptPubKey']
+            if 'type' in scriptPubKey and scriptPubKey['type'] == 'nulldata':
+                assert ('pegout_hex' in scriptPubKey and 'pegout_asm' in scriptPubKey and 'pegout_type' in scriptPubKey and
+                        'pegout_chain' in scriptPubKey and 'pegout_reqSigs' in scriptPubKey and 'pegout_addresses' in scriptPubKey)
+                assert scriptPubKey['pegout_chain'] == self.parentgenesisblockhash
+                assert scriptPubKey['pegout_reqSigs'] == 1
+                assert parent_chain_addr in scriptPubKey['pegout_addresses']
+                pegout_tested = True
+                break
+        assert pegout_tested
+
+    def run_test(self):
+        parent = self.nodes[0]
+        parent2 = self.nodes[1]
+        sidechain = self.nodes[2]
+        sidechain2 = self.nodes[3]
+
+        parent.generate(101)
+        sidechain.generate(101)
+
+        addrs = sidechain.getpeginaddress()
+        addr = parent.validateaddress(addrs["mainchain_address"])
+        print('addrs', addrs)
+        print('addr', addr)
+        txid1 = parent.sendtoaddress(addrs["mainchain_address"], 24)
+        # 10+2 confirms required to get into mempool and confirm
+        parent.generate(1)
+        time.sleep(2)
+        proof = parent.gettxoutproof([txid1])
+        
+        raw = parent.getrawtransaction(txid1)
+        import json
+        print('raw', parent.getrawtransaction(txid1, True))
+
+        print("Attempting peg-in")
+        # First attempt fails the consensus check but gives useful result
+        try:
+            pegtxid = sidechain.claimpegin(raw, proof)
+            raise Exception("Peg-in should not be mature enough yet, need another block.")
+        except JSONRPCException as e:
+            print('ERROR:', e.error)
+            assert("Peg-in Bitcoin transaction needs more confirmations to be sent." in e.error["message"])
+
+        # Second attempt simply doesn't hit mempool bar
+        parent.generate(10)
+        try:
+            pegtxid = sidechain.claimpegin(raw, proof)
+            raise Exception("Peg-in should not be mature enough yet, need another block.")
+        except JSONRPCException as e:
+            assert("Peg-in Bitcoin transaction needs more confirmations to be sent." in e.error["message"])
+
+        # Should fail due to non-witness
+        try:
+            pegtxid = sidechain.claimpegin(raw, proof, get_new_unconfidential_address(parent))
+            raise Exception("Peg-in with non-matching claim_script should fail.")
+        except JSONRPCException as e:
+            print(e.error["message"])
+            assert("Given or recovered script is not a witness program." in e.error["message"])
+
+        # # Should fail due to non-matching wallet address
```

Comment:
Is this here because it should fail due to this if it would not fail due to non-witness (clause above)?

This is very good for clarity, but it doesn't really test if it would fail in that case..

Reply:
That's simply the same test as in https://github.com/ElementsProject/elements/blob/elements-0.14.1/qa/rpc-tests/pegging.py#L233 but commented because I didn't manage to make it work, note here there's a similar one above "Should fail due to non-witness", perhaps that's enough.
Perhaps I should just remove that commented code, but then I will forget about it.
@instagibbs thoughts?

Reply:
was this functionality supposed to be working?